app/src/app/api/users/[user]/organizations/route.ts

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: LXCat team
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import { db } from "@lxcat/database";
+import {
+  badRequestResponse,
+  createdResponse,
+  noContentResponse,
+} from "../../../../../shared/api-responses";
+import { hasAdminRole, hasSession } from "../../../middleware/auth";
+import { zodMiddleware } from "../../../middleware/zod";
+import { RouteBuilder } from "../../../route-builder";
+import { querySchema } from "./schemas";
+
+const router = RouteBuilder
+  .init()
+  .use(hasSession())
+  .use(hasAdminRole())
+  .use(zodMiddleware(querySchema))
+  .post(async (_, ctx) => {
+    if (ctx.parsedParams.body) {
+      const { user: userKey } = ctx.parsedParams.path;
+      await db().setAffiliations(userKey, ctx.parsedParams.body);
+      return createdResponse();
+    } else {
+      return badRequestResponse({
+        body: "Request missing body with organization keys",
+      });
+    }
```

Comment:
Since body is an array, would it be more desirable to accept an empty array and do nothing or throw a bad request response when the array is empty?

Reply:
I think we should either "accept" but return a different status code (e.g. not 201: "Created"), or reject and return a bad request response. As both options essentially boil down to the same thing, it might be easier to handle it in zod (thus resulting in the second option), but perhaps the first option is more "correct"?